core/src/main/java/org/elasticsearch/action/search/SearchPhaseExecutionException.java

```diff
@@ -85,7 +85,13 @@ private static Throwable deduplicateCause(Throwable cause, ShardSearchFailure[]
     @Override
     public RestStatus status() {
         if (shardFailures.length == 0) {
-            // if no successful shards, it means no active shards, so just return SERVICE_UNAVAILABLE
+            // If we have a cause that can report a REST status code then return that.
+            // One such failure occurs when a globally unmapped field is searched with strict search mode on.
+            Throwable cause = unwrapCause();
+            if (cause instanceof ElasticsearchException) {
+                return ((ElasticsearchException) getCause()).status();
+            }
+            // otherwise when no unsuccessful shards, it means no active shards, so just return SERVICE_UNAVAILABLE
```

Comment:
I am a bit worried that we have to modify that logic. Is there a way that we could do without it?

Reply:
I think we can do this but please use `ExceptionsHelper.status(cause)` it handles also when cause is null

Reply:
To be clear, the suggestion being made is to replace the above if statement with:

        if (shardFailures.length == 0) {
            return ExceptionsHelper.status(unwrapCause());
        }

Note that would change the error response of any non-check-field errors from a 503 status to a 500.
This would be a change to existing behaviour.